source/2_platform/rex_opengl/src/resources/shader_program.cpp

```diff
@@ -110,14 +107,34 @@ namespace rex
     GL_CALL(glDeleteShader(fragment_shader_id));
   }
 
+  //-------------------------------------------------------------------------
+  ShaderProgram::ShaderProgram(ShaderProgram&& other) noexcept
+      : m_resource_id(rsl::exchange(other.m_resource_id, 0))
+      , m_uniform_locations(rsl::exchange(other.m_uniform_locations, rsl::unordered_map<rsl::small_stack_string, ShaderUniformLocation>()))
+      , m_bound(rsl::exchange(other.m_bound, false))
+  {
+  }
+
   //-----------------------------------------------------------------------
   ShaderProgram::~ShaderProgram()
   {
     release();
   }
 
+  //-------------------------------------------------------------------------
+  rex::ShaderProgram& ShaderProgram::operator=(ShaderProgram&& other) noexcept
+  {
+    REX_ASSERT(*this == other);
+
+    m_resource_id       = rsl::exchange(other.m_resource_id, 0);
+    m_uniform_locations = rsl::exchange(other.m_uniform_locations, rsl::unordered_map<rsl::small_stack_string, ShaderUniformLocation>());
```

Comment:
this can also just be a move

Reply:
Yes but to be consistent with the other lines within this function I would like to keep it an rsl::exchange.